twisted/internet/inotify.py

```diff
@@ -137,6 +137,7 @@ def _notify(self, filepath, events):
         """
         Callback function used by L{INotify} to dispatch an event.
         """
+        filepath = filepath.asBytesMode()
```

Comment:
So I definitely think we don't need to do this everywhere: it should be enough to do it once in `watch` and once in `ignore`. However, if we do that it becomes impossible to do (2), so I want to see what Adi thinks.